examples/jsm/loaders/LDrawLoader.js

```diff
@@ -1949,9 +1949,17 @@ class LDrawLoader extends Loader {
 	}
 
 	setMaterials( materials ) {
+		this.clearMaterials();
+		this.addMaterials( materials );
+	}
 
+	clearMaterials() {
 		this.materialLibrary = {};
 		this.materials = [];
+	}
+
+	addMaterials( materials ) {
```

Comment:
If `addMaterials()` would be called multiple times (and the `add*()` syntax implies this is intended), then the default materials get added twice. Can you make sure the default triangle and line edge materials are only added once?

Reply:
I'll take a look 👍🏻

Reply:
I've separated out the defaults into their own method, and explicitly called that on load(); I think that makes all the intents clearer.